providers: add an AzureStack stub

darkmuggle opened the pull request:
Stub out AzureStack wire-framing in anticipation of working on AzureStack in the near future. This stub out should allow exploration on the AzureStack platform. 

As a platform AzureStack does not provide either a meta-data service nor does it provide a useful OVF XML the utility of Afterburn is a pretty much a noop for AzureStack.

Signed-off-by: Ben Howard <ben.howard@redhat.com>

lucab commented on src/providers/azurestack/mod.rs:
```diff
@@ -0,0 +1,78 @@
+
+impl MetadataProvider for AzureStack {
+    fn attributes(&self) -> Result<HashMap<String, String>> {
+        let hostname = Self::system_hostname()?;
```
I don't think it is useful to pipe back the local hostname here, we can plainly drop this.

Commit:
providers/mod.rs: move trait noops into MetadataProvider

Generally in all the providers there are trait(s) that are not
implemented or used by the provider; they all had various methods of
noop'ing. This change unifies the noop handling implementing the
required fn in the trait definition itself.

Commit:
AzureStack: implementation stub
